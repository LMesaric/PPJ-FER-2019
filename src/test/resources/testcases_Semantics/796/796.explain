semantic: invalid implicit conversion from `char` to `void` in return
note: 'return a ;' on line 11
