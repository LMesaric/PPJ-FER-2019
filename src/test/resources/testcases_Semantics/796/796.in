<prijevodna_jedinica>
 <prijevodna_jedinica>
  <prijevodna_jedinica>
   <prijevodna_jedinica>
    <prijevodna_jedinica>
     <vanjska_deklaracija>
      <definicija_funkcije>
       <ime_tipa>
        <specifikator_tipa>
         KR_INT 1 int
       IDN 1 jednako
       L_ZAGRADA 1 (
       <lista_parametara>
        <lista_parametara>
         <deklaracija_parametra>
          <ime_tipa>
           <specifikator_tipa>
            KR_CHAR 1 char
          IDN 1 a
        ZAREZ 1 ,
        <deklaracija_parametra>
         <ime_tipa>
          <specifikator_tipa>
           KR_CHAR 1 char
         IDN 1 b
       D_ZAGRADA 1 )
       <slozena_naredba>
        L_VIT_ZAGRADA 1 {
        <lista_naredbi>
         <naredba>
          <naredba_grananja>
           KR_IF 2 if
           L_ZAGRADA 2 (
           <izraz>
            <izraz_pridruzivanja>
             <log_ili_izraz>
              <log_i_izraz>
               <bin_ili_izraz>
                <bin_xili_izraz>
                 <bin_i_izraz>
                  <jednakosni_izraz>
                   <jednakosni_izraz>
                    <odnosni_izraz>
                     <aditivni_izraz>
                      <multiplikativni_izraz>
                       <cast_izraz>
                        <unarni_izraz>
                         <postfiks_izraz>
                          <primarni_izraz>
                           IDN 2 a
                   OP_EQ 2 ==
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          IDN 2 b
           D_ZAGRADA 2 )
           <naredba>
            <slozena_naredba>
             L_VIT_ZAGRADA 2 {
             <lista_naredbi>
              <naredba>
               <naredba_skoka>
                KR_RETURN 3 return
                <izraz>
                 <izraz_pridruzivanja>
                  <log_ili_izraz>
                   <log_i_izraz>
                    <bin_ili_izraz>
                     <bin_xili_izraz>
                      <bin_i_izraz>
                       <jednakosni_izraz>
                        <odnosni_izraz>
                         <aditivni_izraz>
                          <multiplikativni_izraz>
                           <cast_izraz>
                            <unarni_izraz>
                             <postfiks_izraz>
                              <primarni_izraz>
                               BROJ 3 1
                TOCKAZAREZ 3 ;
             D_VIT_ZAGRADA 4 }
           KR_ELSE 4 else
           <naredba>
            <slozena_naredba>
             L_VIT_ZAGRADA 4 {
             <lista_naredbi>
              <naredba>
               <naredba_skoka>
                KR_RETURN 5 return
                <izraz>
                 <izraz_pridruzivanja>
                  <log_ili_izraz>
                   <log_i_izraz>
                    <bin_ili_izraz>
                     <bin_xili_izraz>
                      <bin_i_izraz>
                       <jednakosni_izraz>
                        <odnosni_izraz>
                         <aditivni_izraz>
                          <multiplikativni_izraz>
                           <cast_izraz>
                            <unarni_izraz>
                             <postfiks_izraz>
                              <primarni_izraz>
                               BROJ 5 0
                TOCKAZAREZ 5 ;
             D_VIT_ZAGRADA 6 }
        D_VIT_ZAGRADA 7 }
    <vanjska_deklaracija>
     <definicija_funkcije>
      <ime_tipa>
       <specifikator_tipa>
        KR_VOID 9 void
      IDN 9 greska
      L_ZAGRADA 9 (
      <lista_parametara>
       <deklaracija_parametra>
        <ime_tipa>
         <specifikator_tipa>
          KR_INT 9 int
        IDN 9 x
      D_ZAGRADA 9 )
      <slozena_naredba>
       L_VIT_ZAGRADA 9 {
       <lista_deklaracija>
        <deklaracija>
         <ime_tipa>
          <specifikator_tipa>
           KR_CHAR 10 char
         <lista_init_deklaratora>
          <init_deklarator>
           <izravni_deklarator>
            IDN 10 a
           OP_PRIDRUZI 10 =
           <inicijalizator>
            <izraz_pridruzivanja>
             <log_ili_izraz>
              <log_i_izraz>
               <bin_ili_izraz>
                <bin_xili_izraz>
                 <bin_i_izraz>
                  <jednakosni_izraz>
                   <odnosni_izraz>
                    <aditivni_izraz>
                     <multiplikativni_izraz>
                      <cast_izraz>
                       <unarni_izraz>
                        <postfiks_izraz>
                         <primarni_izraz>
                          ZNAK 10 'a'
         TOCKAZAREZ 10 ;
       <lista_naredbi>
        <naredba>
         <naredba_skoka>
          KR_RETURN 11 return
          <izraz>
           <izraz_pridruzivanja>
            <log_ili_izraz>
             <log_i_izraz>
              <bin_ili_izraz>
               <bin_xili_izraz>
                <bin_i_izraz>
                 <jednakosni_izraz>
                  <odnosni_izraz>
                   <aditivni_izraz>
                    <multiplikativni_izraz>
                     <cast_izraz>
                      <unarni_izraz>
                       <postfiks_izraz>
                        <primarni_izraz>
                         IDN 11 a
          TOCKAZAREZ 11 ;
       D_VIT_ZAGRADA 12 }
   <vanjska_deklaracija>
    <deklaracija>
     <ime_tipa>
      <specifikator_tipa>
       KR_INT 14 int
     <lista_init_deklaratora>
      <init_deklarator>
       <izravni_deklarator>
        IDN 14 x
       OP_PRIDRUZI 14 =
       <inicijalizator>
        <izraz_pridruzivanja>
         <log_ili_izraz>
          <log_i_izraz>
           <bin_ili_izraz>
            <bin_xili_izraz>
             <bin_i_izraz>
              <jednakosni_izraz>
               <odnosni_izraz>
                <aditivni_izraz>
                 <multiplikativni_izraz>
                  <cast_izraz>
                   <unarni_izraz>
                    <postfiks_izraz>
                     <primarni_izraz>
                      BROJ 14 4
     TOCKAZAREZ 14 ;
  <vanjska_deklaracija>
   <definicija_funkcije>
    <ime_tipa>
     <specifikator_tipa>
      KR_VOID 15 void
    IDN 15 f3
    L_ZAGRADA 15 (
    KR_VOID 15 void
    D_ZAGRADA 15 )
    <slozena_naredba>
     L_VIT_ZAGRADA 15 {
     <lista_deklaracija>
      <deklaracija>
       <ime_tipa>
        <specifikator_tipa>
         KR_INT 16 int
       <lista_init_deklaratora>
        <init_deklarator>
         <izravni_deklarator>
          IDN 16 x
         OP_PRIDRUZI 16 =
         <inicijalizator>
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        BROJ 16 3
       TOCKAZAREZ 16 ;
     <lista_naredbi>
      <naredba>
       <naredba_skoka>
        KR_RETURN 17 return
        <izraz>
         <izraz_pridruzivanja>
          <log_ili_izraz>
           <log_i_izraz>
            <bin_ili_izraz>
             <bin_xili_izraz>
              <bin_i_izraz>
               <jednakosni_izraz>
                <odnosni_izraz>
                 <aditivni_izraz>
                  <multiplikativni_izraz>
                   <cast_izraz>
                    <unarni_izraz>
                     <postfiks_izraz>
                      <primarni_izraz>
                       IDN 17 x
        TOCKAZAREZ 17 ;
     D_VIT_ZAGRADA 18 }
 <vanjska_deklaracija>
  <definicija_funkcije>
   <ime_tipa>
    <specifikator_tipa>
     KR_INT 20 int
   IDN 20 main
   L_ZAGRADA 20 (
   KR_VOID 20 void
   D_ZAGRADA 20 )
   <slozena_naredba>
    L_VIT_ZAGRADA 20 {
    <lista_deklaracija>
     <lista_deklaracija>
      <lista_deklaracija>
       <deklaracija>
        <ime_tipa>
         <specifikator_tipa>
          KR_INT 21 int
        <lista_init_deklaratora>
         <init_deklarator>
          <izravni_deklarator>
           IDN 21 a
        TOCKAZAREZ 21 ;
      <deklaracija>
       <ime_tipa>
        <specifikator_tipa>
         KR_CHAR 22 char
       <lista_init_deklaratora>
        <init_deklarator>
         <izravni_deklarator>
          IDN 22 i
         OP_PRIDRUZI 22 =
         <inicijalizator>
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      <postfiks_izraz>
                       <primarni_izraz>
                        ZNAK 22 'a'
       TOCKAZAREZ 22 ;
     <deklaracija>
      <ime_tipa>
       <specifikator_tipa>
        KR_CHAR 23 char
      <lista_init_deklaratora>
       <init_deklarator>
        <izravni_deklarator>
         IDN 23 e
        OP_PRIDRUZI 23 =
        <inicijalizator>
         <izraz_pridruzivanja>
          <log_ili_izraz>
           <log_i_izraz>
            <bin_ili_izraz>
             <bin_xili_izraz>
              <bin_i_izraz>
               <jednakosni_izraz>
                <odnosni_izraz>
                 <aditivni_izraz>
                  <multiplikativni_izraz>
                   <cast_izraz>
                    <unarni_izraz>
                     <postfiks_izraz>
                      <primarni_izraz>
                       ZNAK 23 'b'
      TOCKAZAREZ 23 ;
    <lista_naredbi>
     <lista_naredbi>
      <naredba>
       <izraz_naredba>
        <izraz>
         <izraz_pridruzivanja>
          <postfiks_izraz>
           <primarni_izraz>
            IDN 25 a
          OP_PRIDRUZI 25 =
          <izraz_pridruzivanja>
           <log_ili_izraz>
            <log_i_izraz>
             <bin_ili_izraz>
              <bin_xili_izraz>
               <bin_i_izraz>
                <jednakosni_izraz>
                 <odnosni_izraz>
                  <aditivni_izraz>
                   <multiplikativni_izraz>
                    <cast_izraz>
                     <unarni_izraz>
                      <postfiks_izraz>
                       <postfiks_izraz>
                        <primarni_izraz>
                         IDN 25 jednako
                       L_ZAGRADA 25 (
                       <lista_argumenata>
                        <lista_argumenata>
                         <izraz_pridruzivanja>
                          <log_ili_izraz>
                           <log_i_izraz>
                            <bin_ili_izraz>
                             <bin_xili_izraz>
                              <bin_i_izraz>
                               <jednakosni_izraz>
                                <odnosni_izraz>
                                 <aditivni_izraz>
                                  <multiplikativni_izraz>
                                   <cast_izraz>
                                    <unarni_izraz>
                                     <postfiks_izraz>
                                      <primarni_izraz>
                                       IDN 25 i
                        ZAREZ 25 ,
                        <izraz_pridruzivanja>
                         <log_ili_izraz>
                          <log_i_izraz>
                           <bin_ili_izraz>
                            <bin_xili_izraz>
                             <bin_i_izraz>
                              <jednakosni_izraz>
                               <odnosni_izraz>
                                <aditivni_izraz>
                                 <multiplikativni_izraz>
                                  <cast_izraz>
                                   <unarni_izraz>
                                    <postfiks_izraz>
                                     <primarni_izraz>
                                      IDN 25 e
                       D_ZAGRADA 25 )
        TOCKAZAREZ 25 ;
     <naredba>
      <izraz_naredba>
       <izraz>
        <izraz_pridruzivanja>
         <log_ili_izraz>
          <log_i_izraz>
           <bin_ili_izraz>
            <bin_xili_izraz>
             <bin_i_izraz>
              <jednakosni_izraz>
               <odnosni_izraz>
                <aditivni_izraz>
                 <multiplikativni_izraz>
                  <cast_izraz>
                   <unarni_izraz>
                    <postfiks_izraz>
                     <postfiks_izraz>
                      <primarni_izraz>
                       IDN 26 greska
                     L_ZAGRADA 26 (
                     D_ZAGRADA 26 )
       TOCKAZAREZ 27 ;
    D_VIT_ZAGRADA 28 }
