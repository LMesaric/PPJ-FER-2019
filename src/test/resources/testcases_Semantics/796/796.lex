KR_INT 1 int
IDN 1 jednako
L_ZAGRADA 1 (
KR_CHAR 1 char
IDN 1 a
ZAREZ 1 ,
KR_CHAR 1 char
IDN 1 b
D_ZAGRADA 1 )
L_VIT_ZAGRADA 1 {
KR_IF 2 if
L_ZAGRADA 2 (
IDN 2 a
OP_EQ 2 ==
IDN 2 b
D_ZAGRADA 2 )
L_VIT_ZAGRADA 2 {
KR_RETURN 3 return
BROJ 3 1
TOCKAZAREZ 3 ;
D_VIT_ZAGRADA 4 }
KR_ELSE 4 else
L_VIT_ZAGRADA 4 {
KR_RETURN 5 return
BROJ 5 0
TOCKAZAREZ 5 ;
D_VIT_ZAGRADA 6 }
D_VIT_ZAGRADA 7 }
KR_VOID 9 void
IDN 9 greska
L_ZAGRADA 9 (
KR_INT 9 int
IDN 9 x
D_ZAGRADA 9 )
L_VIT_ZAGRADA 9 {
KR_CHAR 10 char
IDN 10 a
OP_PRIDRUZI 10 =
ZNAK 10 'a'
TOCKAZAREZ 10 ;
KR_RETURN 11 return
IDN 11 a
TOCKAZAREZ 11 ;
D_VIT_ZAGRADA 12 }
KR_INT 14 int
IDN 14 x
OP_PRIDRUZI 14 =
BROJ 14 4
TOCKAZAREZ 14 ;
KR_VOID 15 void
IDN 15 f3
L_ZAGRADA 15 (
KR_VOID 15 void
D_ZAGRADA 15 )
L_VIT_ZAGRADA 15 {
KR_INT 16 int
IDN 16 x
OP_PRIDRUZI 16 =
BROJ 16 3
TOCKAZAREZ 16 ;
KR_RETURN 17 return
IDN 17 x
TOCKAZAREZ 17 ;
D_VIT_ZAGRADA 18 }
KR_INT 20 int
IDN 20 main
L_ZAGRADA 20 (
KR_VOID 20 void
D_ZAGRADA 20 )
L_VIT_ZAGRADA 20 {
KR_INT 21 int
IDN 21 a
TOCKAZAREZ 21 ;
KR_CHAR 22 char
IDN 22 i
OP_PRIDRUZI 22 =
ZNAK 22 'a'
TOCKAZAREZ 22 ;
KR_CHAR 23 char
IDN 23 e
OP_PRIDRUZI 23 =
ZNAK 23 'b'
TOCKAZAREZ 23 ;
IDN 25 a
OP_PRIDRUZI 25 =
IDN 25 jednako
L_ZAGRADA 25 (
IDN 25 i
ZAREZ 25 ,
IDN 25 e
D_ZAGRADA 25 )
TOCKAZAREZ 25 ;
IDN 26 greska
L_ZAGRADA 26 (
D_ZAGRADA 26 )
KR_RETURN 27 return
BROJ 27 1
TOCKAZAREZ 27 ;
D_VIT_ZAGRADA 28 }
