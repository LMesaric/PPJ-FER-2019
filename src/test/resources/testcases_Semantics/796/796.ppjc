int jednako(char a, char b) {
  if(a==b){
     return 1; 
  }else{
     return 0;
 }
}

 void greska(int x) {
   char a='a';
   return a;
 }

 int x=4;
 void f3(void) {
 int x=3;
 return x;
 }
 
 int main(void){
   int a;
   char i='a';
   char e='b';
   
   a=jednako(i,e);
   greska()
 return 1;
 }