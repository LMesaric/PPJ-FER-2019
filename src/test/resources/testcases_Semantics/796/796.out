<naredba_skoka> ::= KR_RETURN(11,return) <izraz> TOCKAZAREZ(11,;)
